conda_build/utils.py

```diff
@@ -15,6 +15,8 @@
 
 from conda_build import external
 
+import logging
+logger = logging.getLogger('conda-build')
```

Comment:
I don't know what conventions other projects use, but in my projects I follow the convention that every source file uses it's own logger (`logger = logging.getLogger(__name__)`), as suggested in the python logging module documentation.  In this PR, you are only declaring a single logger, which is shared across all modules.  Maybe a brief discussion of the pros/cons of each convention is warranted.